doc-content/jbpm-docs/src/main/asciidoc/Persistence/persistence-datamodel-ref.adoc

```diff
@@ -0,0 +1,240 @@
+[id='persistence-datamodel-ref_{context}']
+= Binary runtime persistence data model
+
+The runtime persistence data is internal. As a general rule, do not access the database tables directly and, most importantly, do not modify it. Changing the runtime state information for process instances without using the {PROCESS_ENGINE} might have unexpected consequences. Use a history log to analyze information about the current or past states of process instances.
+
+In rare cases it might be useful to query internal database tables directly if you have a specific need to do so.
+
+The {PROCESS_ENGINE} uses a binary persistence mechanism, otherwise known as marshalling, which converts the state of the process instance into a binary dataset. The same mechanism is also applied to the session state and any work item states.
+
+When the process instance state is persisted, the {PROCESS_ENGINE} completes two steps:
+
+* First, it transforms the process instance information into a binary blob. For performance reasons, the {PROCESS_ENGINE} uses a custom serialization mechanism and not standard Java serialization.
+* Next, it stores the blob alongside metadata about this process instance. This metadata includes the process instance ID, process ID, and process start date, as well as other data.
+
+Apart from process instance states, the {PROCESS_ENGINE} stores some state information for the session, including the state of timer jobs and the session data for evaluating business rules. The {PROCESS_ENGINE} stores the session state as a separate binary blob, along with metadata that includes the ID of the session. 
+
+To restore the state of a session, reload the session with a particular ID using the `ksession.getId()` method.
+
+Binary datasets that represent process instance states are usually relatively small, because they contain only the minimal execution state of the process instance. For a simple process instance, the state usually includes the nodes that are currently executing and any existing variable values.
+
+.Data model for {PROCESS_ENGINE} persistence
+image::Persistence/jbpm_schema_doc.png[]
+
+The `sessioninfo` entity contains the state of the knowledge session in which the process instance is running.
+
+.SessionInfo
+[cols="1,1,1", options="header"]
+|===
+| Field
+| Description
+| Nullable
```

Comment:
I've never seen "Nullable" before but I'd expect a yes/no value in the column, so confused by the entries of either "NOT NULL" or empty. I assume it means "ability to be null", which is basically a confusing way of saying required vs. optional I think. So imo, I'd suggest to follow what we do in our API docs, which is to change this column to be "Requirement" and then for the column values, enter "Required" instead of "Not Null" and enter "Optional" instead of empty cell. 

@fjtirado what do you think about this? ^^ Can you clarify what exactly is meant by "Nullable"? And suggestions for clearer alternative terms to "Nullable"?

Reply:
"Nullable" does look legit, and "required" and "can be null"a re not exactly the same thing. Many fields are logically required, but are not restricted to NOT NULL on the database level.

https://www.webopedia.com/TERM/N/nullable.html

https://stackoverflow.com/questions/55451494/what-does-nullable-signify-for-a-database

https://dataedo.com/kb/query/sql-server/check-column-nullable

Reply:
@ramendik Ah, interesting. So it's indeed a standard in DB documentation and configs, sounds good. However, can we then please just add "NULL" to the empty fields, so that the options are either "NOT NULL" or "NULL"? This seems expected, and I see that being down in a couple of the examples you've shared as well.

Reply:
I think I'm lacking context, but the opossite of Nullable annotation in Java is NotNull, so I would say Nullable vs NotNull ;)